subsys/pm/device.c

```diff
@@ -24,12 +24,26 @@ const char *pm_device_state_str(enum pm_device_state state)
 	}
 }
 
+static int domain_supported_cb(const struct device *dev, void *context)
+{
+	struct pm_device *pm = dev->pm;
+	enum pm_device_action action = (enum pm_device_action)context;
+
+	if (pm->action_cb != NULL) {
+		(void)pm->action_cb(dev, action);
+	}
+
+	return 0;
+}
+
 int pm_device_state_set(const struct device *dev,
 			enum pm_device_state state)
 {
 	int ret;
 	enum pm_device_action action;
 	struct pm_device *pm = dev->pm;
+	bool domain = atomic_test_bit(&dev->pm->flags,
```

Comment:
are atomics required? isn't a const attribute to be a PD?

Reply:
nope, this is just a flag. The field `domain` is used to inform a a device which domain its belong.
Just to be clear, the flag is used to inform that device itself is a power domain.

Reply:
I don't know if it was clear, but using atomic is necessary because the flags type is `atomic_t`. I'm leveraging this variable to set a domain instead of an additional boolean into the struct.